### PR TITLE
Add homebuyers' education verification file upload #162770124

### PR DESCRIPTION
**Pull request by software-project**
https://www.pivotaltracker.com/story/show/162770124

### CODE REVIEW

**Comment by chrisdolendo** [comment]
@software-project what is this blocked by - is it ready for review even if it is blocked?

**Comment by chrisdolendo** [comment]
@software-project Code looks good to me but when QA-ing I could not find the File Uploads -- see application a0W0P00000GlKfB. Can you confirm?

**Comment by software-project** [comment]
@chrisdolendo is blocked by the other file upload PR. This one should go second, then the first one is reviewed and merged.

**Comment by callaghanc** [comment]
@software-project i've changed the base on this to master, can you now resolve merge conflicts?

**Comment by software-project** [comment]
@callaghanc I'm working on it

**Comment by software-project** [comment]
@akegan @callaghanc I rebased this one of master and tested. Looks good for review.

**Comment by callaghanc** [comment]
great thanks, will review!

**Comment by akegan** [comment]
starting QA on this now

File: spec/e2e/utils/page-util.coffee [hunk 1]
```diff
@@ -45,7 +45,7 @@ PageUtil = {
     # need this for uploading file to sauce labs
     browser.setFileDetector new remote.FileDetector()
     filePath = "#{process.env.PWD}#{filePath}"
-    element.all(By.css('input[type="file"]')).then( (items) ->
+    element.all(By.id("ngf-#{documentType}File")).then( (items) ->
```
Comment (akegan) [inline spec/e2e/utils/page-util.coffee]: what was the motivation behind this change?

Comment (akegan) [reply spec/e2e/utils/page-util.coffee]: oh, was this to be able to specify which upload file button you want to click?

**Comment by akegan** [comment]
highlighting is also gone for file name too long issues:
image

Checking on pref validation now.

**Comment by akegan** [comment]
Two problems:
1. Missing the text: "Please upload a valid document" if a document isn't uploaded
2. Doesn't highlight in red if there's a filename too long error or if there's a missing file. 

Issue wasn't present at the beginning of the loan pre-approval PR so it was most likely introduced in this commit https://github.com/Exygy/sf-dahlia-web/pull/1159/commits/b49e83d4e4d7f7d0562925b9f31c4cce63b7e0c0

**Comment by akegan** [comment]
OK QA looks good! 🤞 we can get tests to pass. They haven't been cooperating this afternoon

**Comment by akegan** [comment]
one of the test errors looks like it could be real, it's related to these changes:
``` Scenario: Navigating through a basic application
  ✔ Given I go to the first page of the "Sale Test Listing" application
  ✔ When I hit the Next button "1" time
  ✔ Then I should be on the "Prerequisites" page of the application
  ✔ And I check complete homebuyers education
  ✔ Then I should see an error on the prerequisites form
[23:52:25] E/launcher - Cannot read property 'sendKeys' of undefined
[23:52:25] E/launcher - TypeError: Cannot read property 'sendKeys' of undefined
    at /home/runner/sf-dahlia-web-full/spec/e2e/utils/page-util.coffee:65:24
    at ManagedPromise.invokeCallback_ (/home/runner/sf-dahlia-web-full/node_modules/selenium-webdriver/lib/promise.js:1376:14)
    at TaskQueue.execute_ (/home/runner/sf-dahlia-web-full/node_modules/selenium-webdriver/lib/promise.js:3084:14)
    at TaskQueue.executeNext_ (/home/runner/sf-dahlia-web-full/node_modules/selenium-webdriver/lib/promise.js:3067:27)
    at asyncRun (/home/runner/sf-dahlia-web-full/node_modules/selenium-webdriver/lib/promise.js:2927:27)
    at /home/runner/sf-dahlia-web-full/node_modules/selenium-webdriver/lib/promise.js:668:7
    at process._tickCallback (internal/process/next_tick.js:109:7)
[23:52:25] E/launcher - Process exited with error code 199```

**Comment by software-project** [comment]
Will take a look

**Comment by software-project** [comment]
@akegan pushed a fix for failing feature test. Fingers crossed ;)

**Pull request merged**